src/main/java/com/labsynch/labseer/chemclasses/rdkit/BBChemStructureService.java

```diff
@@ -0,0 +1,338 @@
+		rdKitStructure.setMolecularFormula(RDKFuncs.calcMolFormula(mol, false, false));
+	}
+
+	@Override
+	public JsonNode getPreprocessorSettings() throws IOException {
+			ObjectMapper objectMapper = new ObjectMapper();
+			JsonNode jsonNode = objectMapper.readTree(propertiesUtilService.getPreprocessorSettings());
+			return jsonNode;
+	}
+
+	@Override
+	public RDKitStructure getRDKitStructureFromProcessService(String molfile) throws CmpdRegMolFormatException {
+		RDKitStructure rdkitStructure = new RDKitStructure();
+
+		// Read the preprocessor settings as json
+		JsonNode jsonNode = null;
+		try{
+			jsonNode = getPreprocessorSettings();
+		} catch (IOException e) {
+			logger.error("Error parsing preprocessor settings json");
+			throw new CmpdRegMolFormatException("Error parsing preprocessor settings json");
+		}
+
+		// Extract the processURL
+		JsonNode urlNode = jsonNode.get("processURL");
+		if(urlNode == null || urlNode.isNull()) {
+			logger.error("Missing preprocessorSettings processURL!!");
+		}
+		String url = urlNode.asText();
+
+		// Get the standardization actions and options
+		ObjectMapper mapper = new ObjectMapper();
+		ObjectNode requestData = mapper.createObjectNode();
+		JsonNode standardizerActions = jsonNode.get("standardizer_actions");
+		ObjectNode options = (ObjectNode) jsonNode.get("process_options");
+		options.put("standardizer_actions", standardizerActions);
+		requestData.put("options", options);
+
+		// Add the structures to the request
+		ArrayNode arrayNode = mapper.createArrayNode();
+		arrayNode.add(molfile);
+		requestData.put("structures", arrayNode);
+
+		// Post to the service and parse the response
+		try {
+			String requestString = requestData.toString();
+			logger.info("requestString: " + requestString);
+			String postResponse = SimpleUtil.postRequestToExternalServer(url, requestString, logger);
+			logger.info("Got response: "+ postResponse);
+
+			// Parse the response json to get the standardized mol
+			ObjectMapper responseMapper = new ObjectMapper();
+			JsonNode responseNode = responseMapper.readTree(postResponse);
+			for (JsonNode responseJsonNode : responseNode)  {
+				JsonNode registrationHashesNode = responseJsonNode.get("registration_hash");
+				String registrationHash = registrationHashesNode.get(0).asText();
+				rdkitStructure.setReg(registrationHashesNode.get(0).asText());
+
+				JsonNode noStereoHashNode = responseJsonNode.get("no_stereo_hash");
+				rdkitStructure.setPreReg(noStereoHashNode.asText());
+
+				JsonNode sdfNode = responseJsonNode.get("sdf");
+				rdkitStructure.setMol(sdfNode.asText());
+
+				JsonNode averageMolWeightNode = responseJsonNode.get("molecular_weight");
+				rdkitStructure.setAverageMolWeight(averageMolWeightNode.asDouble());
+
+				JsonNode exactMolWeightNode = responseJsonNode.get("molecular_weight");
+				rdkitStructure.setExactMolWeight(-1.0);
```

Comment:
Why -1 instead of null?

Reply:
Yeah I was writing this up and its not really required here so I'm removing it.